datasets/ar/parliament/crawler.py

```diff
@@ -1,142 +1,91 @@
-    Returns the extracted text, or None if not found.
-    """
     result = element.xpath(xpath_query)
     return result[0].strip() if result else None
 
 
-def crawl_personal_page(context: Context, person_data: dict):
-    """
-    Crawls a person's individual page and updates their data dictionary.
-    """
-    context.log.debug("Starting crawling personal page", person=person_data["url"])
-    doc = context.fetch_html(person_data["url"], cache_days=1)
+def crawl_personal_page(context: Context, url):
+    context.log.debug("Starting crawling personal page", url=url)
+    doc = context.fetch_html(url, cache_days=1)
 
     # Extract additional details from the personal page
-    person_data["profession"] = _extract_text(
-        doc, './/p[@class="encabezadoProfesion"]/span/text()'
-    )
+    profession = _extract_text(doc, './/p[@class="encabezadoProfesion"]/span/text()')
     birth_date = _extract_text(doc, './/p[@class="encabezadoFecha"]/span/text()')
-    person_data["birth_date"] = h.parse_date(birth_date, FORMATS)
-    person_data["email"] = _extract_text(
-        doc, './/a[starts-with(@href, "mailto:")]/text()'
-    )
+    email = _extract_text(doc, './/a[starts-with(@href, "mailto:")]/text()')
 
+    return profession, birth_date, email
 
-def make_and_emit_person(context: Context, person_data: dict):
-    """
-    Creates a 'Person' entity from the scraped data and emits it to the Context.
-    """
-    if person_data.get("name"):
+
+def crawl(context: Context):
+    path = context.fetch_resource("source.html", context.data_url)
+    context.export_resource(path, HTML, title=context.SOURCE_TITLE)
+
+    with open(path, "r") as fh:
+        doc = html.parse(fh)
+        doc = doc.getroot()
+        doc.make_links_absolute(context.data_url)
+
+    # Find the table containing the deputy data
+    for row in h.parse_html_table(doc.find(".//table")):
+        str_row = h.cells_to_str(row)
+        name_el = row.pop("diputado")
+        link = name_el.xpath(".//a/@href")
+
+        name = str_row.pop("diputado")
+        mandate = str_row.pop("mandato")
+        start_date = h.parse_date(
+            str_row.pop("inicia_mandato"), context.dataset.dates.formats
+        )[0]
+        end_date = h.parse_date(
+            str_row.pop("finaliza_mandato"), context.dataset.dates.formats
+        )[0]
+
+        # Create and emit the person entity
         person = context.make("Person")
-        person.id = context.make_id(person_data["name"])
-        person.add("name", person_data["name"])
-        person.add("alias", person_data["alias"])
+        person.id = context.make_id(name)
+        person.add("name", name)
         person.add("country", "ar")
-        h.apply_name(
-            person,
-            first_name=person_data["first_name"].strip(),
-            last_name=person_data["last_name"].strip(),
-        )
-        person.add("sourceUrl", person_data["url"])
```

Comment:
looks like we've lost sourceUrl an applying first and last name (constructing a correctly-ordered full name). Could you add that back?
https://www.opensanctions.org/statements/Q107840687/?dataset=ar_parliament

Reply:
also alias, although I'm not sure if that's worth any effort because in the case of Ignacio it's the same as his name - so either don't drop it, or only drop it if we're sure it's always the constructed name.

Reply:
after a rewrite like this, I would check each field individually to see if I've changed any behaviour. By comparing the data and/or by going through each person.add in the old and the new code to see that all the ones we want to keep are still there.

Reply:
I don't think original source has aliases, that's why I dropped them